app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -59,6 +62,7 @@
 
     @Override
     public Completable querySearch(@NonNull final String query) {
+        searchStateOnceAndStream.onNext(SEARCH_TRIGGERED());
         return freeSoundSearchService.search(get(query))
                                      .map(DefaultSearchDataModel::toResults)
                                      .doOnSuccess(this::storeValueAndClearError)
```

Comment:
You also need to clear progress whenever the `Completable` terminates - either success or error.